TEST: modin/test/test_partition_api.py is very flaky

### System information
- **Modin version** (`modin.__version__`): ce8f18765dd9201039769b022f92735792d63454

### Describe the problem

modin/test/test_partition_api.py seems to be failing more than 50% of the time.

### Source code / logs
Here's a run with a more informative stack trace: https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true

<details>

<summary>Show stack trace</summary>

```
[gw0] linux -- Python 3.8.13 /usr/share/miniconda/envs/modin/bin/python
worker 'gw0' crashed while running 'modin/test/test_partition_api.py::test_from_partitions[0-index-columns-row_lengths-None]'
__________ test_from_partitions[None-index-columns-row_lengths-None] ___________
[gw1] linux -- Python 3.8.13 /usr/share/miniconda/envs/modin/bin/python

axis = None, index = RangeIndex(start=0, stop=65536, step=1)
columns = Int64Index([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,
            ...
            246, 247, 248, 249, 250, 251, 252, 253, 254, 255],
           dtype='int64', length=512)
row_lengths = [65536], column_widths = None

    @pytest.mark.parametrize("column_widths", [None, "column_widths"])
    @pytest.mark.parametrize("row_lengths", [None, "row_lengths"])
    @pytest.mark.parametrize("columns", [None, "columns"])
    @pytest.mark.parametrize("index", [None, "index"])
    @pytest.mark.parametrize("axis", [None, 0, 1])
    def test_from_partitions(axis, index, columns, row_lengths, column_widths):
        num_rows = 2**16
        num_cols = 2**8
        data = np.random.randint(0, 100, size=(num_rows, num_cols))
        df1, df2 = pandas.DataFrame(data), pandas.DataFrame(data)
        expected_df = pandas.concat([df1, df2], axis=1 if axis is None else axis)
    
        index = expected_df.index if index == "index" else None
        columns = expected_df.columns if columns == "columns" else None
        row_lengths = (
            None
            if row_lengths is None
            else [num_rows, num_rows]
            if axis == 0
            else [num_rows]
        )
        column_widths = (
            None
            if column_widths is None
            else [num_cols]
            if axis == 0
            else [num_cols, num_cols]
        )
    
        if Engine.get() == "Ray":
            if axis is None:
                futures = [[put_func(df1), put_func(df2)]]
            else:
                futures = [put_func(df1), put_func(df2)]
        if Engine.get() == "Dask":
            if axis is None:
                futures = [put_func([df1, df2], hash=False)]
            else:
                futures = put_func([df1, df2], hash=False)
        actual_df = from_partitions(
            futures,
            axis,
            index=index,
            columns=columns,
            row_lengths=row_lengths,
            column_widths=column_widths,
        )
>       df_equals(expected_df, actual_df)

modin/test/test_partition_api.py:147: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
modin/pandas/test/utils.py:583: in df_equals
    df2 = to_pandas(df2)
modin/utils.py:396: in to_pandas
    return modin_obj._to_pandas()
modin/pandas/dataframe.py:2877: in _to_pandas
    return self._query_compiler.to_pandas()
modin/core/storage_formats/pandas/query_compiler.py:258: in to_pandas
    return self._modin_frame.to_pandas()
modin/core/dataframe/pandas/dataframe/dataframe.py:114: in run_f_on_minimally_updated_metadata
    result = f(self, *args, **kwargs)
modin/core/dataframe/pandas/dataframe/dataframe.py:2771: in to_pandas
    df = self._partition_mgr_cls.to_pandas(self._partitions)
modin/core/dataframe/pandas/partitioning/partition_manager.py:630: in to_pandas
    retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
modin/core/dataframe/pandas/partitioning/partition_manager.py:630: in <listcomp>
    retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
modin/core/dataframe/pandas/partitioning/partition_manager.py:630: in <listcomp>
    retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
modin/core/dataframe/pandas/partitioning/partition.py:125: in to_pandas
    dataframe = self.get()
modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py:71: in get
    return ray.get(self.oid)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/ray/_private/client_mode_hook.py:105: in wrapper
    return func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

object_refs = [ObjectRef(4f4ef6205ce35f90ffffffffffffffffffffffff0100000001000000)]

    @PublicAPI
    @client_mode_hook(auto_init=True)
    def get(
        object_refs: Union[ray.ObjectRef, List[ray.ObjectRef]],
        *,
        timeout: Optional[float] = None,
    ) -> Union[Any, List[Any]]:
        """Get a remote object or a list of remote objects from the object store.
    
        This method blocks until the object corresponding to the object ref is
        available in the local object store. If this object is not in the local
        object store, it will be shipped from an object store that has it (once the
        object has been created). If object_refs is a list, then the objects
        corresponding to each object in the list will be returned.
    
        Ordering for an input list of object refs is preserved for each object
        returned. That is, if an object ref to A precedes an object ref to B in the
        input list, then A will precede B in the returned list.
    
        This method will issue a warning if it's running inside async context,
        you can use ``await object_ref`` instead of ``ray.get(object_ref)``. For
        a list of object refs, you can use ``await asyncio.gather(*object_refs)``.
    
        Args:
            object_refs: Object ref of the object to get or a list of object refs
                to get.
            timeout (Optional[float]): The maximum amount of time in seconds to
                wait before returning.
    
        Returns:
            A Python object or a list of Python objects.
    
        Raises:
            GetTimeoutError: A GetTimeoutError is raised if a timeout is set and
                the get takes longer than timeout to return.
            Exception: An exception is raised if the task that created the object
                or that created one of the objects raised an exception.
        """
        worker = global_worker
        worker.check_connected()
    
        if hasattr(worker, "core_worker") and worker.core_worker.current_actor_is_asyncio():
            global blocking_get_inside_async_warned
            if not blocking_get_inside_async_warned:
                logger.warning(
                    "Using blocking ray.get inside async actor. "
                    "This blocks the event loop. Please use `await` "
                    "on object ref with asyncio.gather if you want to "
                    "yield execution to the event loop instead."
                )
                blocking_get_inside_async_warned = True
    
        with profiling.profile("ray.get"):
            is_individual_id = isinstance(object_refs, ray.ObjectRef)
            if is_individual_id:
                object_refs = [object_refs]
    
            if not isinstance(object_refs, list):
                raise ValueError(
                    "'object_refs' must either be an object ref "
                    "or a list of object refs."
                )
    
            # TODO(ujvl): Consider how to allow user to retrieve the ready objects.
            values, debugger_breakpoint = worker.get_objects(object_refs, timeout=timeout)
            for i, value in enumerate(values):
                if isinstance(value, RayError):
                    if isinstance(value, ray.exceptions.ObjectLostError):
                        worker.core_worker.dump_object_store_memory_usage()
                    if isinstance(value, RayTaskError):
                        raise value.as_instanceof_cause()
                    else:
>                       raise value
E                       ray.exceptions.WorkerCrashedError: The worker died unexpectedly while executing this task. Check python-core-worker-*.log files for more information.

/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/ray/worker.py:1811: WorkerCrashedError
----------------------------- Captured stderr call -----------------------------
(_apply_func pid=18940) *** SIGBUS received at time=1652990090 on cpu 1 ***
(_apply_func pid=18940) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f8e81541a83 while already in AbslFailureSignalHandler()
(_apply_func pid=18940) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f8e81541a77 while already in AbslFailureSignalHandler()
(_apply_func pid=18940) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f8e81541a77 while already in AbslFailureSignalHandler()
(_apply_func pid=18940) PC: @     0x7f8e81541a83  (unknown)  (unknown)
(_apply_func pid=18940)     @     0x7f8e81720420  (unknown)  (unknown)
(_apply_func pid=18940)     @          0x1555540  (unknown)  (unknown)
(_apply_func pid=18940)     @     0x7f8e7ffe32d0  (unknown)  (unknown)
(_apply_func pid=18940)     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=18940) [2022-05-19 19:54:50,427 E 18940 20130] logging.cc:325: *** SIGBUS received at time=1652990090 on cpu 1 ***
(_apply_func pid=18940) [2022-05-19 19:54:50,473 E 18940 20130] logging.cc:325: PC: @     0x7f8e81541a83  (unknown)  (unknown)
(_apply_func pid=18940) [2022-05-19 19:54:50,473 E 18940 20130] logging.cc:325:     @     0x7f8e81720420  (unknown)  (unknown)
(_apply_func pid=18940) [2022-05-19 19:54:50,474 E 18940 20130] logging.cc:325:     @          0x1555540  (unknown)  (unknown)
(_apply_func pid=18940) [2022-05-19 19:54:50,482 E 18940 20130] logging.cc:325:     @     0x7f8e7ffe32d0  (unknown)  (unknown)
(_apply_func pid=18940) [2022-05-19 19:54:50,490 E 18940 20130] logging.cc:325:     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=18940) Fatal Python error: Bus error
(_apply_func pid=18940) 
2022-05-19 19:57:32,677	WARNING worker.py:1382 -- A worker died or was killed while executing a task by an unexpected system error. To troubleshoot the problem, check the logs for the dead worker. RayTask ID: 8ba3d7d9be0fa5853a1aa960c22d83e85a92569e01000000 Worker ID: 0f8c6b01cad0a66e3b40ca3[488](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:489)917da37f184b376a84b64c686a8534 Node ID: 2931cadc26414f5ea47bef9c3d5749316ff052d0c11c2da789556061 Worker IP address: 10.1.0.107 Worker port: 38935 Worker PID: 18940
(_apply_func pid=18941) *** SIGBUS received at time=1652990255 on cpu 0 ***
(_apply_func pid=18941) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f06a2eeea83 while already in AbslFailureSignalHandler()
(_apply_func pid=18941) PC: @     0x7f06a2eeea77  (unknown)  (unknown)
(_apply_func pid=18941)     @     0x7f06a30cd420  (unknown)  (unknown)
(_apply_func pid=18941)     @          0x1555540  (unknown)  (unknown)
(_apply_func pid=18941) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f06a2eeea77 while already in AbslFailureSignalHandler()
(_apply_func pid=18941) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f06a2eeea83 while already in AbslFailureSignalHandler()
(_apply_func pid=18941)     @     0x7f06a19902d0  (unknown)  (unknown)
(_apply_func pid=18941)     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=18941) [2022-05-19 19:57:35,644 E 18941 20570] logging.cc:325: *** SIGBUS received at time=1652990255 on cpu 0 ***
(_apply_func pid=18941) [2022-05-19 19:57:35,646 E 18941 20570] logging.cc:325: PC: @     0x7f06a2eeea77  (unknown)  (unknown)
(_apply_func pid=18941) [2022-05-19 19:57:35,646 E 18941 20570] logging.cc:325:     @     0x7f06a30cd420  (unknown)  (unknown)
(_apply_func pid=18941) [2022-05-19 19:57:35,660 E 18941 20570] logging.cc:325:     @          0x1555540  (unknown)  (unknown)
(_apply_func pid=18941) [2022-05-19 19:57:35,680 E 18941 20570] logging.cc:325:     @     0x7f06a19902d0  (unknown)  (unknown)
(_apply_func pid=18941) [2022-05-19 19:57:35,682 E 18941 20570] logging.cc:325:     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=18941) Fatal Python error: Bus error
(_apply_func pid=18941) 
2022-05-19 19:57:36,133	WARNING worker.py:1382 -- A worker died or was killed while executing a task by an unexpected system error. To troubleshoot the problem, check the logs for the dead worker. RayTask ID: 8283a89682fd056c08da4c0a69b55961bc30166201000000 Worker ID: cecdcd7348e24a6c193c39264ede3aa418cb9d725226c14f144ea4a1 Node ID: 2931cadc26414f5ea47bef9c3d5749316ff052d0c11c2da789556061 Worker IP address: 10.1.0.107 Worker port: 36987 Worker PID: 18941
(_apply_func pid=20581) *** SIGBUS received at time=1652990260 on cpu 0 ***
(_apply_func pid=20581) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f08d7b1da83 while already in AbslFailureSignalHandler()
(_apply_func pid=20581) PC: @     0x7f08d7b1da77  (unknown)  (unknown)
(_apply_func pid=20581)     @     0x7f08d7cfc420  (unknown)  (unknown)
(_apply_func pid=20581)     @          0x1555540  1785800976  (unknown)
(_apply_func pid=20581)     @     0x7f08d65bf2d0  (unknown)  (unknown)
(_apply_func pid=20581) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f08d7b1da83 while already in AbslFailureSignalHandler()
(_apply_func pid=20581)     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=20581) [2022-05-19 19:57:40,234 E 20581 20730] logging.cc:325: *** SIGBUS received at time=1652990260 on cpu 0 ***
(_apply_func pid=20581) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f08d7b1da77 while already in AbslFailureSignalHandler()
(_apply_func pid=20581) [2022-05-19 19:57:40,236 E 20581 20730] logging.cc:325: PC: @     0x7f08d7b1da77  (unknown)  (unknown)
(_apply_func pid=20581) [2022-05-19 19:57:40,236 E 20581 20730] logging.cc:325:     @     0x7f08d7cfc420  (unknown)  (unknown)
(_apply_func pid=20581) [2022-05-19 19:57:40,237 E 20581 20730] logging.cc:325:     @          0x1555540  1785800976  (unknown)
(_apply_func pid=20581) [2022-05-19 19:57:40,241 E 20581 20730] logging.cc:325:     @     0x7f08d65bf2d0  (unknown)  (unknown)
(_apply_func pid=20581) [2022-05-19 19:57:40,243 E 20581 20730] logging.cc:325:     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=20581) Fatal Python error: Bus error
(_apply_func pid=20581) 
2022-05-19 19:57:40,446	WARNING worker.py:1382 -- A worker died or was killed while executing a task by an unexpected system error. To troubleshoot the problem, check the logs for the dead worker. RayTask ID: 7ceb837939e8c9e1791087d027e55509a07c00a001000000 Worker ID: 182b2d82d83ef69d4cbcc1e86c2fdbf277eb4b639df4f889285[492](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:493)d2 Node ID: 2931cadc26414f5ea47bef9c3d57[493](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:494)16ff052d0c11c2da789556061 Worker IP address: 10.1.0.107 Worker port: 38981 Worker PID: 20581
(_apply_func pid=20580) *** SIGBUS received at time=16[529](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:530)90262 on cpu 1 ***
(_apply_func pid=20580) PC: @     0x7f31f9a15a77  (unknown)  (unknown)
(_apply_func pid=20580)     @     0x7f31f9bf4420  (unknown)  (unknown)
(_apply_func pid=20580)     @          0x1555[540](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:541)  1200818064  (unknown)
(_apply_func pid=20580)     @     0x7f31f84b72d0  (unknown)  (unknown)
(_apply_func pid=20580) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f31f9a15a83 while already in AbslFailureSignalHandler()
(_apply_func pid=20580)     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=20580) [2022-05-19 19:57:42,155 E 20580 20800] logging.cc:325: *** SIGBUS received at time=1652990262 on cpu 1 ***
(_apply_func pid=20580) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f31f9a15a83 while already in AbslFailureSignalHandler()
(_apply_func pid=20580) [2022-05-19 19:57:42,159 E 20580 20800] logging.cc:325: PC: @     0x7f31f9a15a77  (unknown)  (unknown)
(_apply_func pid=20580) [2022-05-19 19:57:42,159 E 20580 20800] logging.cc:325:     @     0x7f31f9bf4420  (unknown)  (unknown)
(_apply_func pid=20580) [failure_signal_handler.cc : 331] RAW: Signal 7 raised at PC=0x7f31f9a15a77 while already in AbslFailureSignalHandler()
(_apply_func pid=20580) [2022-05-19 19:57:42,161 E 20580 20800] logging.cc:325:     @          0x1555540  1200818064  (unknown)
(_apply_func pid=20580) [2022-05-19 19:57:42,166 E 20580 20800] logging.cc:325:     @     0x7f31f84b72d0  (unknown)  (unknown)
(_apply_func pid=20580) [2022-05-19 19:57:42,167 E 20580 20800] logging.cc:325:     @ 0x7de907894810c083  (unknown)  (unknown)
(_apply_func pid=20580) Fatal Python error: Bus error
(_apply_func pid=20580) 
--------------------------- Captured stderr teardown ---------------------------
2022-05-19 19:57:42,382	WARNING worker.py:1382 -- A worker died or was killed while executing a task by an unexpected system error. To troubleshoot the problem, check the logs for the dead worker. RayTask ID: 50a9936fa1[545](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:546)592cd682b5b0c42b228085801de01000000 Worker ID: 6da3ce2fbf4f4a7b31d[551](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:552)f4f66b413b6095db1c15b59a570349e469 Node ID: 2931cadc26414f5ea47bef9c3d5749316ff052d0c11c2da789[556](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:557)061 Worker IP address: 10.1.0.107 Worker port: 43179 Worker PID: 20[580](https://github.com/modin-project/modin/runs/6513946096?check_suite_focus=true#step:11:581)
_______________________ modin/test/test_partition_api.py _______________________
```
</details>

and others with warnings like `worker 'gw0' crashed while running 'modin/test/test_partition_api.py::test_from_partitions[0-index-columns-row_lengths-None]':

- https://github.com/modin-project/modin/runs/6512553686?check_suite_focus=true
- https://github.com/modin-project/modin/runs/6560888517?check_suite_focus=true

### Solutions

I don't have any good ideas. It looks like Ray workers are crashing during the test. Was this test taking a long time before it became flaky?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -257,7 +257,10 @@ jobs:
       - run: python -m pytest modin/test/storage_formats/base/test_internals.py
       - run: python -m pytest modin/test/storage_formats/pandas/test_internals.py
       - run: python -m pytest modin/test/test_envvar_npartitions.py
-      - run: python -m pytest -n 2 modin/test/test_partition_api.py
+      # TODO(https://github.com/modin-project/modin/issues/4550): this test is
+      # slow (~140 sec), but using 2 cores usually causes ray worker crashes.
+      # Fix the crashes and increase cores to 2.
+      - run: python -m pytest modin/test/test_partition_api.py
       - run: python -m pytest modin/test/test_utils.py
       - run: python -m pytest asv_bench/test/test_utils.py
       - run: python -m pytest modin/test/exchange/dataframe_protocol/base

--- a/docs/release_notes/release_notes-0.16.0.rst
+++ b/docs/release_notes/release_notes-0.16.0.rst
@@ -22,7 +22,7 @@ Key Features and Updates
 * Developer API enhancements
   *
 * Update testing suite
-  *
+  * TEST-#4550: Reduce test_partition_api pytest threads to deflake it (#4551)
 * Documentation improvements
   * DOCS-#4552: Change default sphinx language to en to fix sphinx >= 5.0.0 build (#4553)
 * Dependencies

--- a/docs/release_notes/release_notes-0.16.0.rst
+++ b/docs/release_notes/release_notes-0.16.0.rst
@@ -22,7 +22,7 @@ Key Features and Updates
 * Developer API enhancements
   *
 * Update testing suite
-  * TEST-#4550: Reduce test_partition_api pytest threads to deflake it (#4551)
+  * TEST-#4508: Reduce test_partition_api pytest threads to deflake it (#4551)
 * Documentation improvements
   * DOCS-#4552: Change default sphinx language to en to fix sphinx >= 5.0.0 build (#4553)
 * Dependencies

